sdk/core/azure-core/src/main/java/com/azure/core/util/BinaryData.java

```diff
@@ -593,6 +594,21 @@ public InputStream toStream() {
         return new ByteArrayInputStream(this.data);
     }
 
+    /**
+     * Returns a read-only {@link ByteBuffer} representation of this {@link BinaryData}.
+     * <p>
+     * Attempting to mutate the returned {@link ByteBuffer} will throw a {@link ReadOnlyBufferException}.
+     *
+     * <p><strong>Get a read-only ByteBuffer from the BinaryData</strong></p>
+     *
+     * {@codesnippet com.azure.util.BinaryData.toReadOnlyByteBuffer}
+     *
+     * @return A read-only {@link ByteBuffer} representing the {@link BinaryData}.
+     */
+    public ByteBuffer toReadOnlyByteBuffer() {
```

Comment:
Do we want to shorten this to `toByteBuffer` and just document that it is read-only?

@srnagar, @JonathanGiles

Reply:
I like that suggestion

Reply:
Changed to `toByteBuffer` with documentation explaining that it is read-only.